### PR TITLE
Issue 868 combo box string options

### PR DESCRIPTION
**Pull request by mahmoud-zakria-sociomantic**
For #868

### CODE REVIEW

File: src/ComboBox/index.jsx [hunk 1]
```diff
@@ -476,7 +494,8 @@ export default class ComboBox extends Component
                     selection.length && `(${selection.length} items selected)`;
             }
         }
-
+        const options = normalizeOptions( this.props.options ) ||
```
Comment (daniel-martic-sociomantic) [inline src/ComboBox/index.jsx]:
```suggestion

        const options = normalizeOptions( this.props.options ) ||
```

File: src/ComboBox/utils.js [hunk 2]
```diff
@@ -71,6 +80,7 @@ function prefixOptions( options = [], prefix )
  */
 function removePrefix( str, prefix )
 {
+
```
Comment (daniel-martic-sociomantic) [inline src/ComboBox/utils.js]:
```suggestion
```

Comment (mahmoud-zakria-sociomantic) [reply src/ComboBox/utils.js]: Sure

File: src/ComboBox/index.jsx [hunk 3]
```diff
@@ -477,6 +495,8 @@ export default class ComboBox extends Component
             }
         }
 
+        const options = normalizeOptions( this.props.options ) ||
```
Comment (damian-rodriguez-sociomantic) [inline src/ComboBox/index.jsx]: `getDerivedStateFromProps()` will set the normalized options as `this.state.options`, then it wouldn't be necessary to call this function here again

File: src/ComboBox/index.jsx [hunk 4]
```diff
@@ -447,7 +466,6 @@ export default class ComboBox extends Component
             isMultiselect,
             isReadOnly,
             isSearchable,
-            options,
```
Comment (damian-rodriguez-sociomantic) [inline src/ComboBox/index.jsx]: This shouldn't be removed, "normalized options" should be passed directly from `getDerivedStateFromProps()`

File: src/ComboBox/utils.js [hunk 5]
```diff
@@ -52,10 +52,19 @@ function prefixOptions( options = [], prefix )
...
             ...option,
             id : addPrefix( option.id, prefix ),
         };
+
+        const returnedOptions = typeof option !== 'string' ?
+            structuredOptions : flatOptions;
+
+        return returnedOptions;
     } );
 }
```
Comment (damian-rodriguez-sociomantic) [inline src/ComboBox/utils.js]: these changes wouldn't be necessary as we should pass already a "Normalized Options" array to `prefixOptions()`

**Comment by mahmoud-zakria-sociomantic** [comment]
@damian-rodriguez-sociomantic Thanks man for the catching, I appreciate the explanation and support you gave to me.
All done now, can you double check?

**Pull request merged**